cmake/kokkos_compiler_id.cmake

```diff
@@ -147,49 +147,66 @@ ENDIF()
+SET(KOKKOS_MESSAGE_TEXT "${KOKKOS_MESSAGE_TEXT}\n    MSVC               19.29 or higher")
 SET(KOKKOS_MESSAGE_TEXT "${KOKKOS_MESSAGE_TEXT}\nCompiler: ${KOKKOS_CXX_COMPILER_ID} ${KOKKOS_CXX_COMPILER_VERSION}\n")
 
-IF(KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
-  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 4.0.0)
+IF(KOKKOS_CXX_COMPILER_ID STREQUAL Clang AND NOT Kokkos_ENABLE_CUDA)
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 8.0.0)
+    MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
+  ENDIF()
+ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL Clang AND Kokkos_ENABLE_CUDA)
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 10.0.0)
     MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
   ENDIF()
 ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL GNU)
-  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 5.3.0)
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 8.2.0)
     MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
   ENDIF()
 ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL Intel)
-  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 17.0.0)
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 19.0.5)
+    MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
+  ENDIF()
+ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL IntelLLVM AND NOT Kokkos_ENABLE_SYCL)
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 2021.1.1)
     MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
   ENDIF()
 ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL IntelLLVM AND Kokkos_ENABLE_SYCL)
-  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 2022.0.0)
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 2022.0.0) #FIXME 2022.2.0
     MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
   ENDIF()
 ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA)
-  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 9.2.88)
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 11.0.0)
     MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
   ENDIF()
   SET(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "Kokkos turns off CXX extensions" FORCE)
 ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)
   # Note that ROCm 4.5 reports as version 4.4
-  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 4.4.0)
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 4.4.0) #FIXME ROCm 5.2
     MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
   ENDIF()
-ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL PGI)
-  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 17.4)
+ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL PGI OR KOKKOS_CXX_COMPILER_ID STREQUAL NVHPC)
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 22.3)
     MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
   ENDIF()
   # Treat PGI internally as NVHPC to simplify handling both compilers.
   # Before CMake 3.20 NVHPC was identified as PGI, nvc++ is
   # backward-compatible to pgc++.
   SET(KOKKOS_CXX_COMPILER_ID NVHPC CACHE STRING INTERNAL FORCE)
+ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL "MSVC")
+  IF(KOKKOS_CXX_COMPILER_VERSION VERSION_LESS 19.29)
+    MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
+  ENDIF()
+ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL XL OR KOKKOS_CXX_COMPILER_ID STREQUAL XLClang)
+  MESSAGE(FATAL_ERROR "${KOKKOS_MESSAGE_TEXT}")
```

Comment:
The list does not explicitly mention IBM XL
Should we say it is not supported?
Also should we have a flag to turn the error into a warning and continue processing?

Reply:
See 30fc3d617.